rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+
+## Placement
+
+**The placment describes the relationship between doc-comments and the expression they are documenting.**
+
+- Doc-comments are placed before the documentable node. Only `WHITESPACES` are allowed in between. ([Examples](#basic-examples)) 
+    - `WHITESPACES` are: `[\n \r ' ' \t]`.
+
+- The documentation present before the `attribute path` describes the body of the attribute. ([Examples](#Attributes))
+    - In case placement is ambiguous, the one closer to the body has higher precedence. ([Examples](#ambiguous-placement))
+         
+- All partial functions of a curried lambda can share the same placement with the outermost lambda. ([Examples](#partial-lambda-functions))
+  
+> Note: Research of the RFC Sheperds Team showed that this allows for intuitive placements like are already done in nixpkgs.
+
+### Examples
+
+#### Basic examples
+
+Only whitespaces between the `documentable node` and the `doc-comment`
+
+```nix
+/**Doc for anonymous lambda function*/
+↓
+x: x;
+```
+
+```nix
+listToAttrs [
+  { name = "foo"; value = /**Documentation for '1'*/1; }
+]
+```
+
+#### Attributes 
+
+It is allowed to write the documentation before the attribute instead of placing it right before the body.
+
+```nix
+/**Doc for lambda function bound to a variable*/
+           ↓
+assigned = x: x;
+```
+
+```nix
+{
+    /**This documents the specialisation `map (x: x)` */
+          ↓
+    foo = map (x: x);          
+}
+```
+
+#### Ambiguous placement
+
+```nix
+/**Doc B*/
+int = /**Doc A*/1;
+
+# Documentation is 'Doc A' because it is directly next to the documentable body.
+```
+
+#### Dynamic attribute
+
+```nix
+{
+    /** Documentation for '2' */
+                                   ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Dynamic attribute
+```
+
+#### Partial lambda functions
+
+```nix
+/**Generic doc for all partial lambda functions*/
+↓1 ↓2          ↓3
+x: ({y, ...}:  z: x + y) * z;
+
+# All partial functions can share the same placement with the outermost lambda.
+# 1 -> Lambda Value `x: ({y, ...}: z: x + y) * z` 
+# 2 -> Lambda Value `({y, ...}: z: x + y) * z`    
+# 3 -> Lambda Value `z: x + y) * z`               
+```
+
+# Decisions
+
+Each subsection here contains a decision along with arguments and counter-arguments for (+) and against (-) that decision.
+
+> Note: This RFC does not promote any tool; It focuses only on the format, not implementation. At the same time, this RFC must be technically feasible.
+
+## `/**` to start a doc-comment
+
+**Observing**: Doc-comments' outer format should be a distinctive subset of regular comments. Nevertheless, it should allow native writing without an IDE or editor support.
+
+**Considering**: `/** {content} */` where `{content}` is the inner format.
```

Comment:
I thought **Considering** was for providing requirements and argumentation.

Reply:
Uh okay i'll change that